lib/http-invocation.js

```diff
@@ -177,20 +163,23 @@ HttpInvocation.prototype.createRequest = function() {
   var ctorAccepts = null;
   var query;
   var i;
+  var auth = this.auth;
 
   // initial url is the format
   req.url = this.base + method.getFullPath();
 
   // add auth if it is set
-  if (this.username && this.password) {
-    req.auth = {
-      // request defaults to sending auth immidiately, which works for
-      // Basic auth, but doesn't work for Digest auth
-      // Note: these options are ignored if the URL still contains credentials
-      sendImmediately: false,
-      user: this.username,
-      pass: this.password,
-    };
+  if (auth) {
+    req.auth = {};
+    if(auth.username && auth.password) {
+      req.auth.username = auth.username;
+      req.auth.password = auth.password;
+    }
+    if('sendImmediately' in auth) {
```

Comment:
I find it weird that one branch passes potentially any value type, and the other passes a boolean. Should the else branch set `req.auth.sendImmediately = undefined` or `null` then?